ui/pages/confirmations/components/confirm/header/header.tsx

```diff
@@ -14,15 +18,29 @@ import {
   TextVariant,
 } from '../../../../../helpers/constants/design-system';
 import { getAvatarNetworkColor } from '../../../../../helpers/utils/accounts';
+import { useConfirmContext } from '../../../context/confirm';
 import useConfirmationNetworkInfo from '../../../hooks/useConfirmationNetworkInfo';
 import useConfirmationRecipientInfo from '../../../hooks/useConfirmationRecipientInfo';
+import { Confirmation } from '../../../types/confirm';
 import HeaderInfo from './header-info';
+import { WalletInitiatedHeader } from './wallet-initiated-header';
 
 const Header = () => {
   const { networkImageUrl, networkDisplayName } = useConfirmationNetworkInfo();
   const { senderAddress: fromAddress, senderName: fromName } =
     useConfirmationRecipientInfo();
 
+  const { currentConfirmation } = useConfirmContext<Confirmation>();
+
+  if (currentConfirmation?.type === TransactionType.tokenMethodTransfer) {
+    const isWalletInitiated =
+      (currentConfirmation as TransactionMeta).origin === 'metamask';
+
+    if (isWalletInitiated) {
+      return <WalletInitiatedHeader />;
```

Comment:
Minor, as an alternate to returning an enitrely separate component here, is there much overlap between the two headers such that it's worth just dynamically including the back button etc?

Reply:
I think the components are sufficiently different to warrant separate components